Prefetching-Library/android_prefetching_lib/src/main/java/nl/vu/cs/s2group/nappa/PrefetchingLib.java

```diff
@@ -230,6 +300,16 @@ public static void registerActivity(String activityName) {
             poolExecutor.schedule(() -> {
                 PrefetchingDatabase.getInstance().activityDao().insert(activityData);
                 updateActivityMap(PrefetchingDatabase.getInstance().activityDao().getListActivity());
+
+                // Add LiveData observers
+                ActivityNode currentNode = activityGraph.getCurrent();
+                Long activityId = activityMap.get(currentActivityName);
+                if (activityId == null)
+                    throw new IllegalArgumentException("Unknown activity " + currentActivityName);
+                addSessionAggregateObserver(currentNode, activityId);
+                addActivityExtraObserver(currentNode, activityId);
+                addVisitTimePerActivityObserver(currentNode);
+                addAUrlCandidateObserver(currentNode, activityId);
```

Comment:
Moved from the set current activity method. Similar to the init method, this was failing due to not running in the main thread.  Exceptions outside the main thread do not crashes the app.